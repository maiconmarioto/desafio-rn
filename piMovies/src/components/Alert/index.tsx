import React from 'react';
import AwesomeAlert from 'react-native-awesome-alerts';

interface Props {
  showAlert: boolean;
  title: string;
  message: string;
  confirmText: string;
  onConfirm: () => void;
  confirmColor?: string;
  cancelText?: string;
  showCancel?: boolean;
  onCancel?: () => void;
}

const Alert: React.FC<Props> = (props) => {
  const {
    showAlert,
    title,
    message,
    confirmText,
    onConfirm,
    confirmColor,
    cancelText,
    showCancel,
    onCancel,
  } = props;

  const messageStyle = {textAlign: 'center'};

  return (
    <AwesomeAlert
      show={showAlert}
      showProgress={false}
      title={title}
      message={message}
      messageStyle={messageStyle}
      closeOnTouchOutside={true}
      closeOnHardwareBackPress={true}
      showCancelButton={showCancel}
      showConfirmButton={true}
      cancelText={cancelText || 'Cancelar'}
      confirmText={confirmText || 'Confirmar'}
      confirmButtonColor={confirmColor || '#312e38'}
      onCancelPressed={onCancel}
      onConfirmPressed={onConfirm}
    />
  );
};

export default Alert;
