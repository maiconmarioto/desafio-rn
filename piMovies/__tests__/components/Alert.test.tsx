import React from 'react';
import renderer from 'react-test-renderer';

import Alert from '../../src/components/Alert';

test('renders correctly', () => {
  const alert = renderer
    .create(
      <Alert
        confirmText="Tentar novamente"
        showCancel={false}
        onConfirm={() => {}}
        showAlert={true}
        title="Oh NÃ£o"
        message="Teste"
      />,
    )
    .toJSON();
  expect(alert).toMatchSnapshot();
});
