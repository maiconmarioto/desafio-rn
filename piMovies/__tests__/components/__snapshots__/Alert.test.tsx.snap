// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "rgba(52,52,52,0.5)",
            "height": 1334,
            "position": "absolute",
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "maxWidth": "80%",
          "padding": 10,
          "transform": Array [
            Object {
              "scale": 0.30000000000000004,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#626262",
                "fontSize": 18,
                "paddingHorizontal": 15,
                "paddingVertical": 5,
              },
              undefined,
            ]
          }
        >
          Oh NÃ£o
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#7b7b7b",
                "fontSize": 14,
                "paddingTop": 5,
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          Teste
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginTop": 5,
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 5,
                  "margin": 5,
                  "paddingHorizontal": 10,
                  "paddingVertical": 7,
                },
                Object {
                  "backgroundColor": "#312e38",
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 13,
                  },
                  undefined,
                ]
              }
            >
              Tentar novamente
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
